### PR TITLE
Improve tests

### PR DESCRIPTION
**Pull request by fregante**
Re: https://github.com/fregante/code-tag/pull/1#issuecomment-1191660892
cc @fisker

### CODE REVIEW

File: test.mjs [hunk 1]
```diff
@@ -27,5 +28,24 @@ function testContext({any, html, css, gql, md}, name) {
 			assert.equal(gql`a${'b'}c${1}`, 'abc1');
 			assert.equal(md`a${'b'}c${1}`, 'abc1');
 		});
+
+		it('strings with escapes', () => {
+			assert.equal(any`\\\na${'\\\na'}`, '\\\na\\\na');
```
Comment (fisker) [inline test.mjs]: Why `\\\`

```suggestion
			assert.equal(any`\u{1F600}${'\u{1F600}'}\n${'\n'}`, '😀😀\n\n');
```

Comment (fisker) [reply test.mjs]: Prevent pass wrong raw to `String.raw`

```text
String.raw`\u{1F600}${'\u{1F600}'}\n${'\n'}`
'\\u{1F600}😀\\n\n'
```

Comment (fregante) [reply test.mjs]: Not `\\\` but `\\\n`, it's `\` +`line break`

File: test.mjs [hunk 2]
```diff
@@ -27,5 +28,24 @@ function testContext({any, html, css, gql, md}, name) {
...
+			assert.equal(html`\\\na${'\\\na'}`, '\\\na\\\na');
```
Comment (fisker) [inline test.mjs]: Why not assert  `html`, `css`  ... functions self are the same, so we only need  test one function.

```js
assert.equal(html, any);
assert.equal(css, any);
...
```

File: test.mjs (hunk 1 above)
Comment (fregante) [reply test.mjs]: I added a combined emoji check.



> Prevent pass wrong raw to `String.raw`

I think `'\\'` already does that

**Review by fisker — APPROVED** [review]
👍

File: test.mjs [hunk 3]
```diff
@@ -12,20 +13,26 @@ testContext(esm, 'esm');
...
-			assert.equal(any`a${'b'}c${1}`, 'abc1');
-			assert.equal(html`a${'b'}c${1}`, 'abc1');
-			assert.equal(css`a${'b'}c${1}`, 'abc1');
-			assert.equal(gql`a${'b'}c${1}`, 'abc1');
-			assert.equal(md`a${'b'}c${1}`, 'abc1');
+		it('stringifiable objects', () => {
+			const stringifiableObject = new Date();
+			const stringifiedObject = String(stringifiableObject);
+
+			assert.equal(any`${stringifiableObject}`, stringifiedObject, 'Interpolate objects with toString');
```
Comment (fisker) [inline test.mjs:35]:
```suggestion
			const date = new Date();
			const dateString = String(date);

			assert.equal(any`${date}`, dateString, 'Interpolate objects with toString');
```

The names are not so good, is there anything in js can't "stringify"?

Comment (fregante) [reply test.mjs:35]: I just wanted to make it clear that I don’t really care it’s a date. My first attempt was `{toString: () => 'a'}`

Comment (fregante) [reply test.mjs:35]: > The names are not so good, is there anything in js can't "stringify"?

```js
String(Object.create(null))
// TypeError: No default value
```

**Pull request merged**